torch/serialization.py

```diff
@@ -312,6 +312,9 @@ def persistent_id(obj):
                     location,
                     obj.size(),
                     view_metadata)
+        # See Note [Serialize opaque tensors]
+        elif isinstance(obj, torch.device) and obj.type == 'xla':
+            return ('OpaqueDevice', 'cpu')
```

Comment:
I still don't understand why you are special casing the XLA device object on serialization to be an "opaque cpu device". You're just losing information doing this; seems weird to me.